src/overlays/actors/ovl_Bg_Spot15_Rrbox/z_bg_spot15_rrbox.c

```diff
@@ -21,45 +34,340 @@ const ActorInit Bg_Spot15_Rrbox_InitVars = {
     (ActorFunc)BgSpot15Rrbox_Update,
     (ActorFunc)BgSpot15Rrbox_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot15_Rrbox/func_808B3960.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot15_Rrbox/func_808B39E8.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 500, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
+
+Vec3f D_808B45C4[] = {
+    { 770.0f, 1490.0f, -299.0f },
+    { 770.0f, 1550.0f, -299.0f },
+};
+
+Vec3f D_808B45DC[] = { { 29.99f, 0.01f, -29.99f },
+                       { -29.99f, 0.01f, -29.99f },
+                       { -29.99f, 0.01f, 29.99f },
+                       { 29.99f, 0.01f, 29.99f },
+                       { 0.0f, 0.01f, 0.0f } };
+
+s32 D_808B4618[] = { 0, 0 };
+
+void func_808B3960(BgSpot15Rrbox* this, GlobalContext* globalCtx, s32* arg2, s32 flags) {
+    s32 pad;
+    s32 sp30 = 0;
+    u32 pad2;
+
+    DynaPolyInfo_SetActorMove(&this->dyna, flags);
+    DynaPolyInfo_Alloc(arg2, &sp30);
+
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, sp30);
+
+    if (this->dyna.dynaPolyId == 0x32) {
+        osSyncPrintf("Warning : move BG 登録失敗(%s %d)(name %d)(arg_data 0x%04x)\n", "../z_bg_spot15_rrbox.c", 171,
+                     this->dyna.actor.id, this->dyna.actor.params);
+    }
+}
+
+void func_808B39E8(Vec3f* arg0, Vec3f* arg1, f32 arg2, f32 arg3) {
+    arg0->x = ((arg1->z * arg2) + (arg1->x * arg3));
+    arg0->y = arg1->y;
+    arg0->z = ((arg1->z * arg3) - (arg1->x * arg2));
+}
+
+void func_808B3A34(BgSpot15Rrbox* this) {
+    this->bgId = 50;
+}
+
+s32 func_808B3A40(BgSpot15Rrbox* this, GlobalContext* globalCtx) {
+    DynaPolyActor* temp_v0;
+
+    temp_v0 = DynaPolyInfo_GetActor(&globalCtx->colCtx, this->bgId);
+
+    if (temp_v0 != 0 && Math3D_Dist2DSq(temp_v0->actor.posRot.pos.x, temp_v0->actor.posRot.pos.z,
+                                        this->dyna.actor.posRot.pos.x, this->dyna.actor.posRot.pos.z) < 0.01f) {
+        return true;
+    }
+    return false;
+}
+
+s32 func_808B3AAC(BgSpot15Rrbox* this, GlobalContext* globalCxt) {
+    s16 phi_v0;
```

Comment:
I think this function's return value is used as a bool, so you can change it to return true/false and remove the `!= 0` from the check when it's called